### ISSUE TITLE
Unify AndroidManifest for all build flavors

### ISSUE THREAD

**Issue by SecUpwN** [SecUpwN/Android-IMSI-Catcher-Detector#720]
Currently we have two separate `AndroidManifest.xml`: One for the normal build flavor (located in `/AIMSICD/src/main/`) and one for the system build flavor (located in `/AIMSICD/src/system/`). I am afraid that when moving on with development, the `AndroidManifest.xml` for *system* builds could get outdated since we tend to work on our main normal build flavor. Question: Could we enhance our app so that we only have *one* unified `AndroidManifest.xml` located under `/AIMSICD/src/main/` for *all build flavors*? In my eyes, this would ensure we don't have to maintain two files with the same purpose.

**Comment by larsgrefer** [comment]
We could have 3 source roots at the moment:
- `src/system/` (only for the system flavor)
- `src/normal/` (only for the normal flavor, currently not used)
- `src/main/` (for all flavors)

for the system build-flavor the files `src/main/AndroidManifest.xml` and `src/system/AndroidManifest.xml` are merged during the build. So changes in `src/main/AndroidManifest.xml` also affect the system flavor and `src/system/AndroidManifest.xml` only contains declarations specific for the system flavor.

More information about Manifest merging: http://tools.android.com/tech-docs/new-build-system/user-guide/manifest-merger

TL;DR:
normal = `src/normal/` + `src/main/`
system = `src/system/` + `src/main/`

**Comment by SecUpwN** [comment]
>We could have 3 source roots at the moment...

We could. But we don't have. Isn't it better to have *one* source root? What is the preferred way?

>normal = `src/normal/` + `src/main/`
>system = `src/system/` + `src/main/`

Looks like our project gets too complex when adding separate roots and essentially maintaining separate file structures. I'd prefer if we have one root and one `AndroidManifest.xml`, then then when selecting the build flavor everything should be devided apart cleanly. Or is that something impossible?

**Comment by larsgrefer** [comment]
Multiple flavors cause multiple source roots

>then then when selecting the build flavor

what exactly do you mean here?
>everything should be devided apart cleanly

It currently is divided apart cleanly into different source roots

**Comment by SecUpwN** [comment]
>what exactly do you mean here?

I thought that when building our app Android Studio asks which build flavor to build.

>It currently is divided apart cleanly into different source roots

Ok, so do we need an improvement on the source roots? A joined `AndroidManifest.xml` would be better.

**Comment by larsgrefer** [comment]
> I thought that when building our app Android Studio asks which build flavor to build.

Yes you could either choose the flavor in Android Studio or on the commandline by invoking either the `assembleNormal` or `assembleSystem` gradle tasks

>Do we need an improvement on source roots? 

I don't think so
>Guess a joined `AndroidManifest.xml` would be better.

Why? We have split them on purpose so we don't have all the system-permissions in the normal app

**Comment by SecUpwN** [comment]
>We have split them on purpose so we don't have all the system-permissions in the normal app

Thanks for clarifying, I was just wondering why we have to maintain files and if we couldn't unify those in order to avoid the one for system permissions to get outdated (e.g. it lacks the recently introduced `RECEIVE_BOOT_COMPLETED`). Or do permissions added to our main `AndroidManifest.xml` also get transferred to the one for system permissions?

**Comment by larsgrefer** [comment]
>Or do permissions added to our main AndroidManifest.xml also get transferred to the one for system permissions?

Yes, the manifests are merged during the build.

You can find the merged manifests in the `build/intermediates/manifests/full` folder

for more information look here: https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/720#issuecomment-167991237
